### PR TITLE
fix(DataSpreadsheet): added null check

### PR DESCRIPTION
**Pull request by AustinGitHub**
Add null check

Closes # https://github.com/carbon-design-system/ibm-products/issues/5270

#### What did you change?

I added null check 

#### How did you test and verify your work?

I verified using comparing how the logic works for when cell is being deleted in `handleCellDeletion.js` it is then set to `null` so I added that same check here.

### CODE REVIEW

**Comment by AustinGitHub** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx [hunk 1]
```diff
@@ -291,7 +291,11 @@ export let DataSpreadsheet = React.forwardRef(
                   activeCellCoordinates?.column
                 ]
               : null;
-          setActiveCellContent(activeCellFullData.render('Cell'));
+          if (activeCellFullData) {
+            setActiveCellContent(activeCellFullData.render('Cell'));
+          } else {
+            setActiveCellContent(null);
+          }
```
Comment (makafsal) [inline packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:298]: I think we can replace this with the Nullish coalescing operator (??) and optional chaining (.?).
`setActiveCellContent(activeCellFullData?.render('Cell') ?? null);`

**Review by makafsal — DISMISSED** [review]
Looks fine

File: packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx (hunk 1 above)
Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx:298]: Sure

Comment (AustinGitHub) [reply packages/ibm-products/src/components/DataSpreadsheet/DataSpreadsheet.tsx]: Done, thanks!

**Review by emyarod — APPROVED** [review]
looks good to me!

**Comment by AustinGitHub** [comment]
Hi @makafsal can you re-review so this PR can be merged in, thanks!

**Comment by AustinGitHub** [comment]
Thanks Matthew!

**Comment by AustinGitHub** [comment]
Coming back to this PR because we moved up package upgrade to 2.40 and I think the original null check I had should've been applied instead. I am seeing this issue on our table where when we edit it is setting it to empty (which is not what is expected)


https://github.com/carbon-design-system/ibm-products/assets/10100397/4669c2a6-4425-4b25-8cd7-52f9c192219c



I checked the code import and the issue is it is doing this where it sets it to false because of the condition rather than the actual render object. 

https://github.com/carbon-design-system/ibm-products/assets/10100397/592f2508-0d70-4fea-8d40-ddead1abad88

I will open an issue for this and make a fix since this is regression

**Pull request merged**